src/panels/config/lovelace/dashboards/dialog-lovelace-dashboard-detail.ts

```diff
@@ -0,0 +1,293 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import "../../../../components/ha-dialog";
+import "../../../../components/ha-icon-input";
+import { HomeAssistant } from "../../../../types";
+import {
+  LovelaceDashboard,
+  LovelaceDashboardMutableParams,
+  LovelaceDashboardCreateParams,
+} from "../../../../data/lovelace";
+import { LovelaceDashboardDetailsDialogParams } from "./show-dialog-lovelace-dashboard-detail";
+import { PolymerChangedEvent } from "../../../../polymer-types";
+import { HaSwitch } from "../../../../components/ha-switch";
+
+@customElement("dialog-lovelace-dashboard-detail")
+export class DialogLovelaceDashboardDetail extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() private _params?: LovelaceDashboardDetailsDialogParams;
+  @property() private _urlPath!: LovelaceDashboard["url_path"];
+  @property() private _confMode!: LovelaceDashboard["mode"];
+  @property() private _showSidebar!: boolean;
+  @property() private _sidebarIcon!: string;
+  @property() private _sidebarTitle!: string;
+  @property() private _requireAdmin!: LovelaceDashboard["require_admin"];
+
+  @property() private _error?: string;
+  @property() private _submitting = false;
+
+  public async showDialog(
+    params: LovelaceDashboardDetailsDialogParams
+  ): Promise<void> {
+    this._params = params;
+    this._error = undefined;
+    if (this._params.dashboard) {
+      this._urlPath = this._params.dashboard.url_path || "";
+      this._confMode = this._params.dashboard.mode || "storage";
+      this._showSidebar = !!this._params.dashboard.sidebar;
+      this._sidebarIcon = this._params.dashboard.sidebar?.icon || "";
+      this._sidebarTitle = this._params.dashboard.sidebar?.title || "";
+      this._requireAdmin = this._params.dashboard.require_admin || false;
+    } else {
+      this._urlPath = "";
+      this._confMode = "storage";
+      this._showSidebar = true;
+      this._sidebarIcon = "";
+      this._sidebarTitle = "";
+      this._requireAdmin = false;
+    }
+    await this.updateComplete;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._params) {
+      return html``;
+    }
+    const urlInvalid = !/^[a-zA-Z0-9_-]+$/.test(this._urlPath);
```

Comment:
Can we also do a check that there is not currently a panel that uses this ? (maybe future PR)